api/auth/aws/aws.go

```diff
@@ -108,7 +109,7 @@ func (a *AWSAuth) Login(ctx context.Context, client *api.Client) (*api.Secret, e
 			}
 			pkcs7 := strings.TrimSpace(resp)
 			loginData["pkcs7"] = pkcs7
-		} else {
+		} else if a.signatureType == identityType {
```

Comment:
Curious where you saw that the `identity` type uses SHA-1? I was looking at [our docs](https://www.vaultproject.io/api-docs/auth/aws#signature) and [instance-identity-documents.html#verify-iid](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instance-identity-documents.html#verify-iid) and only see usage of SHA-256. 

Somehow I've managed to not look at the code for this auth method (so far :)) so trying to learn a bit.

Reply:
I believe it is the AWS certificate -- it's signed with sha1.